discovery-infra/start_discovery.py

```diff
@@ -353,6 +431,23 @@ def nodes_flow(client, cluster_name, cluster, image_path):
                 utils.config_etc_hosts(cluster_info)
                 utils.wait_for_cvo_available()
 
+def _get_libvirt_nodes_from_tf_state(network_name, tf_state):
+    nodes = _extract_nodes(tf_state, network_name, consts.NodeRoles.MASTER)
+    nodes.update(_extract_nodes(tf_state, network_name, consts.NodeRoles.WORKER))
```

Comment:
Since the function is called for both roles, why call it twice for each role? Why the function does not return all roles?

Reply:
There are two sections in tfstate - one is called "master", the other "worker". I need to do the same sequence of steps for each of the sections. Putting this in one function would mean duplicating the code, instead of applying the same code to one section at a time.

Reply:
OK